### ISSUE TITLE
Error: Pluginus: plugin "X" not found as dependency for "Y"

### ISSUE THREAD

**Issue by dgilperez** [asd-xiv/pluginus#3]
Hi! 

I am experiencing an issue that seems to be related to the loading order of dependencies. 

The situation is: 

```
PluginA {
  depend: ["PluginB"],
}

PluginB {
  depend: ["PluginA"],
}
```

That causes the error mentioned in the title. Having a quick look at the code, I see that this piece of code could be the issue: 

```
sort((a, b) => {
  const aHasB = has(b.name, a.depend)
  const bHasA = has(a.name, b.depend)
  if (!aHasB && !bHasA) {
    return a.depend.length > b.depend.length ? 1 : -1
  }
  return bHasA ? -1 : 1
}),
```

While it can be argued that the code should be decoupled or architectured in a different way, I wanted to raise the issue in case this is a known limitation, there is a workaround, or this is something that need addressing at this level.

**Comment by andreidmt** [comment]
Yes, you cant have circular dependencies. If A, to be instantiated and fulfill it's responsibility, needs something from B and B needs something from A it doesn't really make sense. (I always forget add a custom throw error for circular dependencies)

The way I've solved this in the past is to move some part of the logic of A, or A & B, in a third C Plugin and use it as a dependency for A and B.

The ides is that there must be a set of "transformations" that can run in sequence in order to satisfy the usecase, nature demands it :)

**Comment by dgilperez** [comment]
Good thinking, that's in deed the natural order of things :) 

Thanks for the suggestion, I'll take that route. 

Just as a comment, I _think_ I saw the same error without (I _think_) a circular dependency, that was fixed by reducing the number of dependencies on one of the classes. 

``` 
PluginA {
  depend: ["PluginB"],
}

PluginC {
  depend: ["PluginB"],
}

PluginB {
  depend: ["PluginD", "PluginE", "PluginF"],
}
```

I removed `F` (it wasn't needed after all) and voilà. But I can't provide a test case or be sure that there was no circular, that code is already gone ... but I am sure that I fixed one instance of this error by removing an unrelated dependency from one of the lists. The sorting by `.length` may be ... 

Just worth mentioning.
